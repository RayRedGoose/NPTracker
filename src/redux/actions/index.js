export const addProcessName = name => ({
  type: 'ADD_PROCESS_NAME',
  name
})

export const addProcessStep = step => ({
  type: 'ADD_PROCESS_STEP',
  step
})

export const increaseStep = () => ({
  type: 'INCREASE_STEP'
})

export const addUserInfo = info => ({
  type: 'ADD_USER_INFO',
  info
})

export const addParks = parks => ({
  type: 'ADD_PARKS',
  parks
})

export const changeActiveTab = id => ({
  type: 'CHANGE_ACTIVE_TAB',
  id
})
