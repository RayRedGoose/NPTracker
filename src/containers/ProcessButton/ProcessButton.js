import React, { Component } from 'react'
import { Redirect } from 'react-router-dom'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import { addProcessName, addProcessStep } from 'redux/actions'

export class ProcessButton extends Component {
  constructor(props) {
    super(props)
    this.state = {
      isRedirected: false
    }
  }

  redirect = (event) => {
    const { addProcessName, addProcessStep } = this.props
    const type = event.target.id
    addProcessName(type)
    addProcessStep(1)
    this.setState({isRedirected: true})
  }

  render() {
    const { text, type } = this.props

    return (
      (!this.state.isRedirected)
        ? <button onClick={this.redirect} id={type}>{text}</button>
        : <Redirect to={`/${type}`} />
    )
  }
}

export const mapDispatchToProps = dispatch => (
  bindActionCreators({
    addProcessName,
    addProcessStep
  }, dispatch)
)

export default connect(null, mapDispatchToProps)(ProcessButton)
