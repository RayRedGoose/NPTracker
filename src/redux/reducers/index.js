import { combineReducers } from 'redux'
import { process } from './singleReducers/process'
import { user } from './singleReducers/user'

const rootReducer = combineReducers({
  process,
  user
})

export default rootReducer
