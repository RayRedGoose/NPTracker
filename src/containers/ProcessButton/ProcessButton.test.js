import React from 'react';
import { shallow } from 'enzyme';
import { ProcessButton, mapDispatchToProps } from './ProcessButton'
import { addProcessName, addProcessStep } from 'redux/actions'

describe("ProcessButton", () => {
  let button, instance, mockProps
  const addProcessName = jest.fn()
  const addProcessStep = jest.fn()

  beforeEach(() => {
    mockProps = {
      text: "Sign In",
      type: 'login',
      addProcessName,
      addProcessStep
    }

    button = shallow(
      <ProcessButton {...mockProps} />
    )

    instance = button.instance()
  });

  describe("Snapshots", () => {
    it("should match snapshot with all data passed in correctly", () => {
      expect(button).toMatchSnapshot()
    });

    it("should match snapshot if isRedirected state is true", () => {
      button.setState({isRedirected: true})
      expect(button).toMatchSnapshot()
    });
  });

  describe("Event handlers", () => {
    it("should call redirect if button is clicked", () => {
      const spy = jest.spyOn(instance, 'redirect').mockImplementation(jest.fn())
      instance.forceUpdate()

      const mockEvent = {target: {id: 'login'}}
      button.find('button').simulate('click', mockEvent)

      expect(spy).toHaveBeenCalledWith(mockEvent);
    });
  });

  describe("redirect fn", () => {
    let mockEvent

    beforeEach(() => {
      mockEvent = {target: {id: 'login'}}
    });

    it("should call addProcessName props with target id as argument if redirect is called", () => {
      instance.redirect(mockEvent)

      expect(addProcessName).toHaveBeenCalledWith('login')
    });

    it("should call addProcessStep props with 1 as argument if redirect is called", () => {
      instance.redirect(mockEvent)

      expect(addProcessStep).toHaveBeenCalledWith(1)
    });

    it("should change isRedirected to true if redirect is called", () => {
      expect(button.state('isRedirected')).toEqual(false)

      instance.redirect(mockEvent)

      expect(button.state('isRedirected')).toEqual(true)
    });
  });
});

describe("mapDispatchToProps", () => {
  let mockDispatch, mockProps

  beforeEach(() => {
    mockDispatch = jest.fn()
    mockProps = mapDispatchToProps(mockDispatch)
  });

  it("should call dispatch with an addProcessName action when addProcessName is called", () => {
    const type = 'login'

    const actionToDispatch = addProcessName(type)

    const result = mockProps.addProcessName(type)

    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });

  it("should call dispatch with an addProcessStep action when addProcessStep is called", () => {
    const actionToDispatch = addProcessStep(1)

    const result = mockProps.addProcessStep(1)

    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });
});
