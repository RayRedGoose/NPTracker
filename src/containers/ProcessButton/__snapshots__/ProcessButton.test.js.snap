// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProcessButton Snapshots should match snapshot if isRedirected state is true 1`] = `
<Redirect
  to="/login"
/>
`;

exports[`ProcessButton Snapshots should match snapshot with all data passed in correctly 1`] = `
<button
  id="login"
  onClick={[Function]}
>
  Sign In
</button>
`;
